test/e2e/timetable/import-timetable.e2e-spec.ts

```diff
@@ -0,0 +1,311 @@
+
+    expect(
+      (await prisma.ueCourse.count({
+        where: {
+          ue: {
+            code: ue.code,
+            id: ue.id,
+          },
+          semester: semester,
+        },
+      })) == 1,
+    );
+
+    expect(
+      (await prisma.ueCourse.count({
+        where: {
+          ue: {
+            code: ue2.code,
+            id: ue2.id,
+          },
+          semester: semester,
+        },
+      })) == 1,
+    );
+  });
+
+  it('should add the first user to course', async () => {
+    setTimetable(
+      `BEGIN:VEVENT
+      UID:a85c5f9f0a3b3f0364e6ee9d4290ecd8
+      DTSTAMP:20240927T112949Z
+      SUMMARY:${semester.code}_${ue.code}_FR_TRO - CM 
+      DESCRIPTION:TP 
+      DTSTART:20240906T100000
+      DTEND:20240906T120000
+      LOCATION:A001
+      END:VEVENT`.replace(/^\s+/gm, ''),
+    );
+
+    await pactum.spec().post(`/timetable/import/${defaultUrl}`).withBearerToken(users[0].token);
+
+    const prisma = app().get(PrismaService);
+
+    expect(
+      (await prisma.ueCourse.count({
+        where: {
+          students: {
+            some: {
+              id: users[0].id,
+            },
+          },
+        },
+      })) == 1,
+    );
+  });
+
+  it('should add users to an existing course', async () => {
+    setTimetable(
+      `BEGIN:VEVENT
+      UID:a85c5f9f0a3b3f0364e6ee9d4290ecd8
+      DTSTAMP:20240927T112949Z
+      SUMMARY:${semester.code}_${ue.code}_FR_TRO - CM 
+      DESCRIPTION:TP 
+      DTSTART:20240906T100000
+      DTEND:20240906T120000
+      LOCATION:A001
+      END:VEVENT`.replace(/^\s+/gm, ''),
+    );
+
+    for (let i = 0; i < users.length; i++) {
+      await pactum.spec().post(`/timetable/import/${defaultUrl}`).withBearerToken(users[i].token);
+    }
+
+    const prisma = app().get(PrismaService);
+
+    expect(
+      (
+        await prisma.ueCourse.findFirst({
+          where: {
+            ue: {
+              code: ue.code,
+              id: ue.id,
+            },
+            semester: semester,
+          },
+          include: {
+            students: true,
+          },
+        })
+      ).students.length == 4,
+    );
+  });
+
+  it('Should return an error when the url is invalid', async () => {
+    const invalidUrl = encodeURIComponent(`https://monedt.utt.fr/calendrier/invalid.ics`);
+    await pactum
+      .spec()
+      .post(`/timetable/import/${invalidUrl}`)
+      .withBearerToken(users[0].token)
+      .expectAppError(ERROR_CODE.RESSOURCE_UNAVAILABLE, `https://monedt.utt.fr/calendrier/invalid.ics`);
+  });
+
+  it('Should return an error when the service send something of the wrong format', async () => {
+    setTimetable('I am not respecting the .ics file format');
+    await pactum
+      .spec()
+      .post(`/timetable/import/${defaultUrl}`)
+      .withBearerToken(users[0].token)
+      .expectAppError(ERROR_CODE.RESSOURCE_INVALID_TYPE, 'ical');
+  });
+
+  it('Should not create duplicates for the same course occurence', async () => {
```

Comment:
tu as pas déjà testé ça ? (`should not create another course if it already exist`)

Reply:
ah si 😅